### ISSUE TITLE
bug: wds append the splitter twice

### ISSUE THREAD

**Issue by maxwellzh** [webdataset/webdataset#164]
Please have a look at https://github.com/webdataset/webdataset/blob/682b30ee484d719a954554654d2d6baa213f9371/webdataset/compat.py#L96-L108

When input `urls` is string like `data-{000..123).tar`, it seems the wds just append both nodesplitter and workersplitter twice, which results the yield data is less than expected.

**Comment by tmbdev** [comment]
Yes, thanks for catching this. I also just noticed this and fixed it.

TODO: add test case
